apps/frontend/src/libs/helpers/get-first-available-route.ts

```diff
@@ -0,0 +1,26 @@
+import {
+	type NavigationItem,
+	type PermissionGetAllItemResponseDto,
+} from "~/libs/types/types.js";
+
+const getFirstAvailableRoute = (
```

Comment:
instead, let's have another helper and then use it inside the sidebar and protected route
```ts
const getPermittedNavigationItems = (items: NavigationItem[], userPermissions: PermissionGetAllItemResponseDto[]) => {
   return items.filter(({ pagePermissions = [] }) => checkHasPermission(pagePermissions, userPermissions),
   );
}
```